monomer.go

```diff
@@ -72,9 +73,23 @@ type Block struct {
 // Hash returns a unique hash of the block, used as the block identifier
 func (b *Block) Hash() common.Hash {
 	if b.Header.Hash == (common.Hash{}) {
-		// We exclude the tx commitment.
-		// TODO better hashing technique than using Ethereum's.
-		b.Header.Hash = (&ethtypes.Header{
+		ethBlock, err := b.ToEth()
+		if err != nil {
+			// TODO: this panic is quite ugly. Perhaps we should just use Ethereum types everywhere?
```

Comment:
No great suggestions here. One idea could be to embed the Hash() func in a constructor, then expose the calculated value.

But there seem to be a few places where partial `Block` literals are populated, so that could be an annoying refactor.

Is it an idea to just have this Hash function return an error?

Reply:
> One idea could be to embed the Hash() func in a constructor, then expose the calculated value.

This seems like the best approach to me. The constructor could maintain other useful invariants as well. Like you mention, this is more of a heavy refactor, so we can put it off for now. I've created a task for it.

Reply:
@NiloCK